packages/url_launcher/url_launcher_windows/pubspec.yaml

```diff
@@ -13,7 +13,7 @@ flutter:
     implements: url_launcher
     platforms:
       windows:
-        pluginClass: UrlLauncherPlugin
+        pluginClass: UrlLauncherWindows
```

Comment:
This is not great, but I don't have a solution I like better so far. The issue is that the `flutter` tool generates both an `#include` path and a call to the C API based on the value of `pluginClass`, so if it stays as FooPlugin, then the public header has to be called `foo_plugin.h`. But the plugin class's file should match its name, so should be called `foo_plugin.h/cpp`, and it's ugly to have two headers with the same name even if they are in different directories (and the implementation file of the C function would need to move, or have a name that doesn't match its header).

So the best option I can see is to have the `pluginClass` value not actually *be* the plugin class's name, but instead the plugin's name without the `Plugin` part (which isn't quite what I did here since there was already a mismatch with the federated name; in the template it would be `Foo` for a plugin whose implementation is `FooPlugin`). That should be harmless, but it's kind of confusing. We could deprecated `pluginClass` and replace it with a new key (continuing to honor pluginClass) to make it less odd, but I'm not sure it's worth the ecosystem migration (and/or inconsistency, depending on how many plugins migrate).

Reply:
Hrm. Also not in love with this. Will book a quick sync to discuss, then we can summarise here.

Reply:
Summary: we agree that there's no obvious good option 😐  We're going to do this for now, and revisit our overall strategy when we update the plugin template to have the class in separate files. (And if we go a different route there, we can always update our plugins here.)